ast/Trees.h

```diff
@@ -716,6 +719,24 @@ TREE(Send) : public Expression {
     virtual std::string showRaw(const core::GlobalState &gs, int tabs = 0);
     virtual std::string nodeName();
 
+    std::pair<int, int> kwArgsRange() const {
+        auto res = std::make_pair<int, int>(numPosArgs, args.size());
+        if (hasKwSplat()) {
+            res.second = res.second - 1;
+        }
+        return res;
+    }
+
+    // True when there are either keyword args, or a keyword splat.
+    bool hasKwArgs() const {
+        return args.size() > numPosArgs;
+    }
+
+    // True when there is a keyword args splat present. hasKwSplat -> hasKwArgs, but not the other way around.
+    bool hasKwSplat() const {
+        return (args.size() - numPosArgs) & 0x1;
+    }
```

Comment:
I'm fine to make this a follow up if you want it.

I wonder if we should make these into one method that returns a pair of booleans, which would in essence force you to ask for both, reminding you at the call site that you're ignoring one.

Reply:
Let's do this in a follow-up.